valora-inc/wallet
# Fix Android Dismiss Modal Animation

### Description

Updates [react-native-modal](https://www.npmjs.com/package/react-native-modal) to [v13.0.1](https://github.com/react-native-modal/react-native-modal/releases/tag/v13.0.1) which includes a fix from [v11.6.0](https://github.com/react-native-modal/react-native-modal/releases/tag/v11.6.0) to avoid flashing of backdrop on Android.

### Other changes

- Uses new react-native-modal props on modals in: WebViewAndroidBottomSheet.tsx, ShakeForSupport.tsx, CashInBottomSheet.tsx, and VerificationEducationScreen.tsx.
- Updated snapshots with default fields included in react-native-modal updates.

### Tested

Tested locally on iOS and Android.

### How others should test

#### WebViewAndroidBottomSheet
1. Navigate to Dapp Explorer.
2. Open a Dapp.
3. Tap triple dot icon.
4. Tap 'Dismiss'.
5. Observe flash when dismissing is no longer present.

#### ShakeForSupport
1. While the app is open shake the device.
2. Tap the X icon in the shake for support modal.
3. Observe flash when dismissing is no longer present.

#### CashInBottomSheet
1. After creating a new account (mainnet) with no balance the the cash in bottom sheet should be present.
2. Tap the X icon in the cash in bottom sheet.
3. Observe flash when dismissing is no longer present.

#### VerificationEducationScreen
1. Navigate to the phone verification screen: (Rewards --> Connect Phone Number).
2. Tap 'Connect Number'
3. Enter a Phone number and tap 'Start'
4. ReCaptcha - 'I'm not a robot' Check box should be served.
5. Tap Close - screen shout dismiss smoothly without flashes.

### Related issues

- Fixes #2588

### Backwards compatibility

Yes
